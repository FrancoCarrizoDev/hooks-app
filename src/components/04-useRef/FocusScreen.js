import React, {useRef} from 'react'

export const FocusScreen = () => {

    const inputRef = useRef()
    const handleClick = () => {
        inputRef.current.select();
    }
    return (
        <div className="p-5">
            <h1>Focus Screen</h1>
            <hr />
            <input
                ref={inputRef}
                type="text"
                className="form-control"
                placeholder="Su nombre" />
                <button 
                    className="btn btn-outline-primary mt-4"
                    onClick={handleClick}               
                >
                    Focus
                </button>
        </div>
    )
}
