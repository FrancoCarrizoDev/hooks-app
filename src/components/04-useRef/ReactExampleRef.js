import React, {useState} from 'react'
import {MultipleCustomHooks} from '../03-examples/MultipleCustomHooks'

export const ReactExampleRef = () => {

    const [show, setShow] = useState(false)
    return (
        <div className="p-5">
            <h1>ReactExampleRef</h1>
            <hr />

            {show && <MultipleCustomHooks/>}

            <button
                className="btn btn-primary mt-4"
                onClick={() => {
                    setShow(!show)
                }}
            >
                Show/Hide
            </button>
        </div>
    )
}
