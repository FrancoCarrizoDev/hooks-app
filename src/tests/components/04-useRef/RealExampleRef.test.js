import { shallow } from "enzyme"
import { ReactExampleRef } from "../../../components/04-useRef/ReactExampleRef"


describe('Test in RealExample', () => {
    
    const wrapper = shallow(<ReactExampleRef/>)

    test('should show this component', () => {
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false)
    })

    test('should show this component MultipleCustomHooks', () => {
        
        wrapper.find('button').simulate('click')
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true)
    })
    
})
