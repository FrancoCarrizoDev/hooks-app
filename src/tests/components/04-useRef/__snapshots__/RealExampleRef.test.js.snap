// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in RealExample should show this component 1`] = `
<div
  className="p-5"
>
  <h1>
    ReactExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Test in RealExample should show this component MultipleCustomHooks 1`] = `
<div
  className="p-5"
>
  <h1>
    ReactExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
